core/services/vrf/delegate_test.go

```diff
@@ -124,104 +126,106 @@ func TestDelegate(t *testing.T) {
 		logListener = args.Get(0).(log.Listener)
 	}).Return(unsubscribe)
 	require.NoError(t, listener.Start())
-
-	t.Run("valid log", func(t *testing.T) {
-		txHash := cltest.NewHash()
-		reqID := cltest.NewHash()
-
-		vuni.lb.On("WasAlreadyConsumed", mock.Anything, mock.Anything).Return(false, nil)
-		a := cltest.NewAwaiter()
-		vuni.lb.On("MarkConsumed", mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
-			a.ItHappened()
-		}).Return(nil).Once()
-		// Expect a call to check if the req is already fulfilled.
-		vuni.ec.On("CallContract", mock.Anything, mock.Anything, mock.Anything).Return(generateCallbackReturnValues(t), nil)
-
-		// Ensure we queue up a valid eth transaction
-		// Linked to  requestID
-		vuni.txm.On("CreateEthTransaction", mock.AnythingOfType("*gorm.DB"), vuni.submitter, common.HexToAddress(vs.CoordinatorAddress), mock.Anything, uint64(500000), mock.MatchedBy(func(meta *models.EthTxMetaV2) bool {
-			return meta.JobID > 0 && meta.RequestID == reqID && meta.RequestTxHash == txHash
-		})).Once().Return(models.EthTx{}, nil)
-
-		// Send a valid log
-		pk, err := secp256k1.NewPublicKeyFromHex(vs.PublicKey)
-		require.NoError(t, err)
-		logListener.HandleLog(log.NewLogBroadcast(types.Log{
-			// Data has all the NON-indexed parameters
-			Data: append(append(append(append(
-				pk.MustHash().Bytes(),                        // key hash
-				common.BigToHash(big.NewInt(42)).Bytes()...), // seed
-				cltest.NewHash().Bytes()...), // sender
-				cltest.NewHash().Bytes()...), // fee
-				reqID.Bytes()...), // requestID
-			// JobID is indexed, thats why it lives in the Topics.
-			Topics:      []common.Hash{{}, jb.ExternalIDToTopicHash()}, // jobID
-			Address:     common.Address{},
-			BlockNumber: 0,
-			TxHash:      txHash,
-			TxIndex:     0,
-			BlockHash:   common.Hash{},
-			Index:       0,
-			Removed:     false,
-		}))
-		a.AwaitOrFail(t)
-
-		// Ensure we created a successful run.
-		runs, err := vuni.jpv2.Prm.GetAllRuns()
-		require.NoError(t, err)
-		require.Equal(t, 1, len(runs))
-		assert.False(t, runs[0].Errors.HasError())
-		m, ok := runs[0].Meta.Val.(map[string]interface{})
-		require.True(t, ok)
-		_, ok = m["eth_tx_id"]
-		assert.True(t, ok)
-		assert.Len(t, runs[0].PipelineTaskRuns, 0)
-		require.NoError(t, store.DB.Exec(`TRUNCATE eth_txes,pipeline_runs CASCADE`).Error)
```

Comment:
This line would occasionally deadlock with the bptxm

Reply:
Oh good grief, that's a nasty one.

Using the mock is the superior fix, but I believe you could also have fixed by reversing the order, i.e. `TRUNCATE pipeline_runs, eth_txes`